tools/mtmd/clip.cpp

```diff
@@ -3900,6 +4385,209 @@ struct img_tool {
         return true;
     }
 
+    // Bicubic resize function using Pillow's ImagingResample algorithm
+    // Adapted from https://github.com/python-pillow/Pillow/blob/main/src/libImaging/Resample.c
+    static bool resize_bicubic_pillow(const clip_image_u8 & img, clip_image_u8 & dst, int target_width, int target_height) {
```

Comment:
please include some comments to explain the algorithmic/numeric differences between `RESIZE_ALGO_BICUBIC_PILLOW` and `RESIZE_ALGO_BICUBIC`